src/validation.cpp

```diff
@@ -1695,15 +1695,15 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
         return PackageMempoolAcceptResult(package_state_quit_early, {});
     }
 
-    // IsChildWithParents() guarantees the package is > 1 transactions.
-    assert(package.size() > 1);
     // The package must be 1 child with all of its unconfirmed parents. The package is expected to
     // be sorted, so the last transaction is the child.
     const auto& child = package.back();
     std::unordered_set<uint256, SaltedTxidHasher> unconfirmed_parent_txids;
-    std::transform(package.cbegin(), package.cend() - 1,
-                   std::inserter(unconfirmed_parent_txids, unconfirmed_parent_txids.end()),
-                   [](const auto& tx) { return tx->GetHash(); });
+    if (package.size() > 1) {
```

Comment:
I think this check can be dropped, operation should be safe and no-op if package is empty? I'm fine with it either way, just simplifies the code a bit.

Reply:
I'm bad at reading the spec, is that defined behavior? https://en.cppreference.com/w/cpp/algorithm/transform

Reply:
ok I'm convinced, removed